Model/Quote/GetQuote.php

```diff
@@ -47,6 +83,30 @@ public function getQuote(
         } catch (LocalizedException $e) {
             return $this->quoteResultBuilder->createErrorResult($e->getMessage());
         }
+        $shipping = $quote->getExtensionAttributes()?->getShippingAssignments();
+        if(count($shipping) > 0) {
+            $shipping = $shipping[0]->getShipping();
+            $shippingAddress = $shipping->getAddress()?->getCountryId();
+            $selectedShippingMethod = $shipping->getMethod();
+            $methods = $this->extractShippingMethods->extract($quote);
+            if(!is_null($shippingAddress) && is_null($selectedShippingMethod) && count($methods) > 0) {
```

Comment:
Thinking out loud here - but thoughts on adding one more check somewhere to see if the currently selected shipping method exists in the list of available shipping methods? 

This could help guard against situations where there is a selected method but the address has changed and so the selected method is no longer valid. We could then select a new default option.

Reply:
I think it would make more sense to check that at the source of the change, as in put that logic in set address.